### PR TITLE
Filter out temporary indices from DB catalog

### PR DESCRIPTION
**Pull request by smakalias**
## Summary of the changes / Why this improves CrateDB

As we may have leftover temp indexes during an incomplete operation (e.g. shrink),
we want to make sure that the db catalog is still functional and that these indexes
are hidden from the user. 

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: sql/src/main/java/io/crate/metadata/DanglingIndex.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    private static boolean isShrinkedIndex(String indexName) {
+        return indexName.startsWith(SHRINK_PREFIX);
+    }
+
+    private static boolean isBackupIndex(String indexName) {
+        return indexName.startsWith(BACKUP_PREFIX);
+    }
+
+    public static boolean isDanglingIndex(String indexName) {
+        return isShrinkedIndex(indexName) || isBackupIndex(indexName);
```
Comment (seut) [inline sql/src/main/java/io/crate/metadata/DanglingIndex.java]: I'd prefer doing it inverse, like filter everything starting with a `.` beside `IndexParts.PARTITIONED_TABLE_PART`

File: sql/src/test/java/io/crate/integrationtests/DanglingIndicesIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import io.crate.metadata.DanglingIndex;
+import io.crate.testing.UseRandomizedSchema;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.junit.Test;
+
+import static org.hamcrest.core.Is.is;
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 1)
+@UseRandomizedSchema(random = false)
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/DanglingIndicesIntegrationTest.java]: minor: As the schema is given (`doc.t1`) randomized schema don't have to be forbidden.

File: sql/src/main/java/io/crate/metadata/DanglingIndex.java (hunk 1 above)
Comment (seut) [reply sql/src/main/java/io/crate/metadata/DanglingIndex.java]: @mfussenegger @andreidan what's your opinion on that?

Comment (smakalias) [reply sql/src/main/java/io/crate/metadata/DanglingIndex.java]: I like this proposal @seut , pushed changes

**Pull request merged**